tests/meta/clean/test_clean_flow_cells_api.py

```diff
@@ -0,0 +1,246 @@
+
+
+def test_is_directory_older_than_21_days_pass(
+    flow_cell_clean_api_can_be_removed: CleanFlowCellAPI,
+):
+    """Test to check whether a directory is older than 21 days."""
+
+    # GIVEN a clean flow cell api with a flow cell that can be deleted
+
+    # THEN checking whether a given flow cell directory is older than 21 days is TRUE
+    with mock.patch(
+        "time.time",
+        return_value=time.time() + TWENTY_ONE_DAYS_IN_SECONDS,
+    ):
+        assert flow_cell_clean_api_can_be_removed.is_directory_older_than_21_days()
+
+
+def test_is_directory_older_than_21_days_fail(flow_cell_clean_api_can_be_removed: CleanFlowCellAPI):
+    """Test to check whether a directory is older than 21 days."""
+
+    # GIVEN a clean flow cell api with a current time that is set to now.
+
+    # THEN checking whether a given flow cell directory is older than 21 days is FALSE
+    assert not flow_cell_clean_api_can_be_removed.is_directory_older_than_21_days()
+
+
+def test_has_sample_sheet_in_housekeeper(flow_cell_clean_api_can_be_removed: CleanFlowCellAPI):
+    """Test to check whether a flow cell has a sample sheet in housekeeper."""
+
+    # GIVEN a clean flow cell api with a flow cell that has a sample sheet in housekeeper
+
+    # THEN checking whether the flow cell has a sample sheet in housekeeper returns TRUE
+    assert flow_cell_clean_api_can_be_removed.has_sample_sheet_in_housekeeper()
+
+
+def test_get_files_for_flow_cell_bundle(flow_cell_clean_api_can_be_removed: CleanFlowCellAPI):
+    """Test to get files for a flow cell bundle from housekeeper."""
+
+    # GIVEN a clean flow cell api with a flow cell that has files in housekeeper
+
+    # WHEN getting fastq and spring files that are tagged with the flow cell
+    fastq_files: List[
+        File
+    ] = flow_cell_clean_api_can_be_removed.get_files_for_samples_on_flow_cell_with_tag(
+        tag=SequencingFileTag.FASTQ
+    )
+    spring_files: List[
+        File
+    ] = flow_cell_clean_api_can_be_removed.get_files_for_samples_on_flow_cell_with_tag(
+        tag=SequencingFileTag.SPRING
+    )
+    spring_metadata_files: List[
+        File
+    ] = flow_cell_clean_api_can_be_removed.get_files_for_samples_on_flow_cell_with_tag(
+        tag=SequencingFileTag.SPRING_METADATA
+    )
+
+    # THEN fastq and SPRING files are returned
+    assert fastq_files
+    assert spring_files
+    assert spring_metadata_files
+
+
+def test_can_flow_cell_be_deleted(flow_cell_clean_api_can_be_removed: CleanFlowCellAPI):
+    """Test the flow cell can be deleted check."""
+    # GIVEN a flow cell that can be deleted
+
+    # WHEN checking that the flow cell can be deleted
+
+    # THEN the check whether the flow cell can be deleted returns True
+    with mock.patch(
+        "cg.meta.clean.clean_flow_cells.CleanFlowCellAPI.is_directory_older_than_21_days",
+        return_value=True,
+    ):
+        assert flow_cell_clean_api_can_be_removed.can_flow_cell_directory_be_deleted()
```

Comment:
Just reordering the terms to be more clear and compliant with GIVEN, WHEN, THEN structure
```suggestion
    # GIVEN a flow cell that can be deleted
    with mock.patch(
        "cg.meta.clean.clean_flow_cells.CleanFlowCellAPI.is_directory_older_than_21_days",
        return_value=True,
    ):
        # WHEN checking that the flow cell can be deleted
        is_deletable: bool = flow_cell_clean_api_can_be_removed.can_flow_cell_directory_be_deleted()

        # THEN the check whether the flow cell can be deleted returns True
        assert is_deletable
    
```